torch/distributed/rpc/backend_registry.py

```diff
@@ -21,14 +22,65 @@ def register_backend(backend_name, init_backend_handler):
             `_init_rpc()` function is called with a backend.
              This returns the agent.
     """
-    backend_registry = _get_backend_registry()
-    if backend_name in backend_registry:
+    global BackendType
+    if backend_name in BackendType.__members__.keys():
         raise RuntimeError("RPC backend {}: already registered".format(backend_name))
-    backend_registry[backend_name] = init_backend_handler
+    # Create a new enum type, `BackendType`, with extended members.
+    existing_enum_dict = {member.name: member.value for member in BackendType}
+    extended_enum_dict = dict(
+        {backend_name: BackendValue(init_backend_handler=init_backend_handler)},
+        **existing_enum_dict
+    )
+    BackendType = enum.Enum(value="BackendType", names=extended_enum_dict)
+    return BackendType[backend_name]
```

Comment:
Not a fan of this approach. It means that `BackendType.FOO_BAR` might fail depending on when you call it.

What's wrong with using a string? Common strings can still be exposed as constant.

Reply:
I intend to fail in that case.
The failure would be a clear and standalized Python built-in enum module "no attribute" error, clearly indicating users that they should register their custom backend type into the type enum first.